game-headed/src/main/java/org/triplea/game/client/ui/javafx/util/JavaFxSelectionComponentFactory.java

```diff
@@ -323,25 +282,61 @@ public String validValueDescription() {
       noneButton.setToggleGroup(toggleGroup);
       systemButton.setToggleGroup(toggleGroup);
       userButton.setToggleGroup(toggleGroup);
+      toggleGroup.selectedToggleProperty().addListener((observable, oldValue, newValue) -> {
+        if (!userButton.isSelected()) {
+          hostText.clear();
+          portText.clear();
+        }
+      });
+
       getChildren().add(radioPanel);
     }
 
     @Override
     public void save(final SaveContext context) {
       if (noneButton.isSelected()) {
         context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.NONE);
+        context.setValue(proxyHostClientSetting, null);
+        context.setValue(proxyPortClientSetting, null);
       } else if (systemButton.isSelected()) {
         context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_SYSTEM_SETTINGS);
+        context.setValue(proxyHostClientSetting, null);
+        context.setValue(proxyPortClientSetting, null);
         HttpProxy.updateSystemProxy();
       } else {
-        context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_USER_PREFERENCES);
+        final String encodedHost = hostText.getText().trim();
+        final Optional<String> optionalHost = parseHost(encodedHost);
```

Comment:
Side-note: the split/duplication of JavaFx components compared to swing seems to go a bit too far to be sustainable. In other words, more directly, we're suffering from parallel copy/paste. While I understand we intend to do a replacement, I'm thinking about two things:
- it' a medium to long term project to do the replacement. The overhead to sustain this situation over that time is not attractive.
- we likely should have the view model logic extracted from our UI code regardless of JavaFx or Swing. In other words, it seems we're blending our model logic in the view. That blending of concerns is pretty pervasive, there are a few patterns to try and help us there but we have not established yet a really solid one. I've been looking at Martin Fowler's notes on view-models, there are some interesting patterns there like '[presentation-model](https://martinfowler.com/eaaDev/PresentationModel.html)' that could help us decouple concerns.

Reply:
Resolving conversation, extracting this topic to a dedicated issue: https://github.com/triplea-game/triplea/issues/4599